cmd/zdb/zdb.c

```diff
@@ -3023,6 +3027,117 @@ verify_dd_livelist(objset_t *os)
 	return (0);
 }
 
+static char *key_material = NULL;
+
+static boolean_t
+zdb_derive_key(dsl_dir_t *dd, uint8_t *key_out)
+{
+	uint64_t keyformat, salt, iters;
+	int i;
+	unsigned char c;
+
+	zap_lookup(dd->dd_pool->dp_meta_objset, dd->dd_crypto_obj,
```

Comment:
```suggestion
	VERIFY0(zap_lookup(dd->dd_pool->dp_meta_objset, dd->dd_crypto_obj,
```

Let's make this `zap_lookup` and the two below in `ZFS_KEYFORMAT_PASSPHRASE` fatal if they fail.